docs/index/secondary-index-guide.md

```diff
@@ -0,0 +1,192 @@
+  for right columns) and the queries should have filter on that column to improve the filter query
+  performance.
+  
+  SI tables will always be loaded non-lazy way. Once SI table is created, Carbondata's 
+  CarbonOptimizer with the help of CarbonSITransformationRule, transforms the query plan to hit the
+  SI table based on the filter condition or set of filter conditions present in the query.
+  So first level of pruning will be done on SI table as it stores blocklets and main table/parent
+  table pruning will be based on the SI output, which helps in giving the faster query results with
+  better pruning.
+
+  Secondary Index table can be create with below syntax
+
+   ```
+   CREATE INDEX [IF NOT EXISTS] index_name
+   [ON TABLE maintable](index_column)
+   AS
+   'carbondata'
+   TBLPROPERTIES('table_blocksize'='1')
+   ```
+  For instance, main table called **sales** which is defined as
+
+  ```
+  CREATE TABLE sales (
+    order_time timestamp,
+    user_id string,
+    sex string,
+    country string,
+    quantity int,
+    price bigint)
+  STORED AS carbondata
+  ```
+
+  User can create SI table using the Create Index DDL
+
+  ```
+  CREATE INDEX index_sales
+  ON TABLE sales(user_id)
+  AS
+  'carbondata'
+  TBLPROPERTIES('table_blocksize'='1')
+  ```
+**NOTE**:
+ * Secondary Index tables are registered with hive and stored in HiveSERDEPROPERTIES as json formatted
+ value. Maximum characters supported for SERDEPROPERTIES by hive is 4000 characters which cannot be
+ changed.
+ 
+ 
+#### How SI tables are selected
+
+When a user executes a filter query, during query planning phase, CarbonData with help of
+CarbonSITransformationRule, checks if there are any index tables present on the filter column of
+query. If there are any, then filter query plan will be transformed such a way that, execution will
+first hit the corresponding SI table and give input to main table for further pruning.
+
+
+For the main table **sales** and SI table  **index_sales** created above, following queries
+```
+SELECT country, sex from sales where user_id = 'xxx'
+
+SELECT country, sex from sales where user_id = 'xxx' and country = 'INDIA'
+```
+
+will be transformed by CarbonData's CarbonSITransformationRule to query against SI table
+**index_sales** first which will be input to the main table **sales**
+
+
+## Loading data
+
+### Loading data to Secondary Index table(s).
+
+*case1:* When SI table is created and the main table does not have any data. In this case every
+consecutive load will load to SI table once main table data load is finished.
+
+*case2:* When SI table is created and main table already contains some data, then SI creation will
+also load to SI table with same number of segments as main table. There after, consecutive load to
+main table will load to SI table also.
+
+ **NOTE**:
+ * In case of any failure to SI table, then we make the SI table disable by setting a hive serde
```

Comment:
```suggestion
 * In case of any failure to SI table, then we make the SI table disable by setting a hive serde
```
If data loading fails for SI table,